internal/graph/check.go

```diff
@@ -728,7 +728,7 @@ func (c *LocalChecker) checkDirect(parentctx context.Context, req *ResolveCheckR
 							ContextualTuples:     req.GetContextualTuples(),
 							ResolutionMetadata: &ResolutionMetadata{
 								Depth:               req.GetResolutionMetadata().Depth - 1,
-								DatastoreQueryCount: response.GetResolutionMetadata().DatastoreQueryCount,
+								DatastoreQueryCount: req.GetResolutionMetadata().DatastoreQueryCount, // add userset read at the end
```

Comment:
we use the same pattern when resolving TTUs: https://github.com/openfga/openfga/blob/d7f77478e8024f3f299c66d7279f9a0ccd84ba28/internal/graph/check.go#L920-L923